### ISSUE TITLE
BUG: application does not support the current operation on symbolic links

### ISSUE THREAD

**Issue by ashish0kumar** [Flow-Launcher/Flow.Launcher#3011]
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I am using the latest version of Flow Launcher.

### Problem Description

Since, I symlinked the config of flow launcher to my dotfiles flow launcher config. This error is showing on startup: 
System.IO.IOException: This application does not support the current operation on symbolic links.

### To Reproduce

1. Symlink the flow launcher config
2. Run flow launcher at startup

### Flow Launcher Version

1.19.2

### Windows Build Number

10.0.22631.4169

### Error Log

[log file](https://github.com/user-attachments/files/17265550/2024-10-05.txt)

**Comment by taooceros** [comment]
Yeah to provide atomicity of the file operation we do not support symbolic link to prevent potential data corruption.

**Comment by taooceros** [comment]
If more request to this we can consider switching to an embedded database rather than plain text.

**Comment by bropines** [comment]
My programs plugin recursively tried to count symlinks in the start menu, which is why the log weighed 15 GB
![explorer_j9A7w0Dil0](https://github.com/user-attachments/assets/8701d6c0-0000-41ea-84bf-b961fa915c6e)

**Comment by taooceros** [comment]
> My programs plugin recursively tried to count symlinks in the start menu, which is why the log weighed 15 GB ![explorer_j9A7w0Dil0](https://private-user-images.githubusercontent.com/57861007/374702887-8701d6c0-0000-41ea-84bf-b961fa915c6e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAxNzk5MDEsIm5iZiI6MTczMDE3OTYwMSwicGF0aCI6Ii81Nzg2MTAwNy8zNzQ3MDI4ODctODcwMWQ2YzAtMDAwMC00MWVhLTg0YmYtYjk2MWZhOTE1YzZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI5VDA1MjY0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZkYWQ3N2MyMTg1NjllYWIzZDQxNTRmNGRhYjMzOGNkZjQyZGQzYTA0YjczNDgyNjZiYTJiZmFlZjE1OWNiNGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.8swB_slYKbl3xaWdeOb501ftEYqGAQTOkClBYZeIL98)

huh that's a different issue. Could you fire another one? Also could you show some snippet inside the very large log that count symlink?

**Comment by helmesjo** [comment]
@taooceros Wouldn't this be trivial to solve by simply first resolving the path, then call `ReplaceFile()`?

**Comment by taooceros** [comment]
> @taooceros Wouldn't this be trivial to solve by simply first resolving the path, then call `ReplaceFile()`?

I guess it makes sense? I am not quite familiar with symlink.

Are you proposing we resolve the final path and then replace the final file instead of the symlink?

**Comment by helmesjo** [comment]
> Are you proposing we resolve the final path and then replace the final file instead of the symlink?

Correct. If the problem is a path that includes one or more symlinks being passed to one of the functions, it should work to just first resolve it (that is, the resolved path no longer contains any symlink(s), it'll just be a simple absolute path) and then invoke the function.
I don't use C# but just intuitively I would guess that should work.

**Comment by bropines** [comment]
> > My programs plugin recursively tried to count symlinks in the start menu, which is why the log weighed 15 GB ![explorer_j9A7w0Dil0](https://private-user-images.githubusercontent.com/57861007/374702887-8701d6c0-0000-41ea-84bf-b961fa915c6e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAxNzk5MDEsIm5iZiI6MTczMDE3OTYwMSwicGF0aCI6Ii81Nzg2MTAwNy8zNzQ3MDI4ODctODcwMWQ2YzAtMDAwMC00MWVhLTg0YmYtYjk2MWZhOTE1YzZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI5VDA1MjY0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZkYWQ3N2MyMTg1NjllYWIzZDQxNTRmNGRhYjMzOGNkZjQyZGQzYTA0YjczNDgyNjZiYTJiZmFlZjE1OWNiNGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.8swB_slYKbl3xaWdeOb501ftEYqGAQTOkClBYZeIL98)
> 
> huh that's a different issue. Could you fire another one? Also could you show some snippet inside the very large log that count symlink?

Ouch.... I didn't see the notification. I can try to repeat that bug. It's done in a couple of minutes.

**Comment by taooceros** [comment]
@ashish0kumar try this build? https://ci.appveyor.com/project/JohnTheGr8/flow-launcher/builds/51303267/artifacts

**Comment by ashish0kumar** [comment]
@taooceros I tested the build, and I'm now able to symlink files without any issues. Thanks for the fix!
